sources/api/thar-be-updates/src/status.rs

```diff
@@ -0,0 +1,352 @@
+    Available,
+    Staged,
+    Ready,
+}
+
+/// UpdateImage represents a Bottlerocket update image
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct UpdateImage {
+    arch: String,
+    version: semver::Version,
+    variant: String,
+}
+
+impl UpdateImage {
+    pub fn get_version(&self) -> &semver::Version {
+        &self.version
+    }
+}
+
+/// StagedImage represents an Bottlerocket image that is written to a partition set
+#[derive(Debug, Deserialize, Serialize)]
+pub struct StagedImage {
+    image: UpdateImage,
+    next_to_boot: bool,
+}
+
+impl StagedImage {
+    pub(crate) fn set_next_to_boot(&mut self, next_to_boot: bool) {
+        self.next_to_boot = next_to_boot
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub enum CommandStatus {
+    Success,
+    Failed,
+    Unknown,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum UpdateCommand {
+    RefreshUpdate,
+    PrepareUpdate,
+    ActivateUpdate,
+    DeactivateUpdate,
+    GetUpdateStatus,
+}
+
+/// CommandResult represents the result of an issued command
+#[derive(Debug, Deserialize, Serialize)]
+struct CommandResult {
+    cmd_type: UpdateCommand,
+    cmd_status: CommandStatus,
+    timestamp: DateTime<Utc>,
+    exit_status: Option<i32>,
+    stderr: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct UpdateStatus {
+    update_state: UpdateState,
+    available_updates: Vec<update_metadata::Update>,
+    chosen_update: Option<UpdateImage>,
+    active_partition: Option<StagedImage>,
+    staging_partition: Option<StagedImage>,
+    most_recent_command: Option<CommandResult>,
+}
+
+impl Default for UpdateStatus {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl UpdateStatus {
+    /// Initializes the update status
+    pub fn new() -> Self {
+        Self {
+            update_state: UpdateState::Idle,
+            available_updates: vec![],
+            chosen_update: None,
+            active_partition: None,
+            staging_partition: None,
+            most_recent_command: None,
+        }
+    }
+
+    pub fn get_state(&self) -> &UpdateState {
+        &self.update_state
+    }
+
+    pub fn get_chosen_update(&self) -> Option<&UpdateImage> {
+        match &self.chosen_update {
+            Some(update) => Some(&update),
+            None => None,
+        }
+    }
+
+    pub fn get_staging_partition_info(&self) -> Option<&StagedImage> {
+        match &self.staging_partition {
+            Some(partition_info) => Some(&partition_info),
+            None => None,
+        }
+    }
+
+    #[allow(clippy::wrong_self_convention)]
+    /// Transitions update state to Idle
+    pub fn to_idle(&mut self) -> Result<()> {
+        match self.update_state {
+            UpdateState::Idle | UpdateState::Available => {
+                self.update_state = UpdateState::Idle;
+                Ok(())
+            }
+            _ => error::InvalidStateTransition {
```

Comment:
For each of these state transition functions, it might be worth noting what purpose the function serves in the program, which might clarify why certain state transitions are invalid.

For example instead of `/// Transition update state to Idle` perhaps `/// Transition update state to idle after an update has been staged and is available` or whatever the case is.